### PR TITLE
[bug] get_cpus is not supporting cgroup2 #11635

### LINKED ISSUE #11635: [bug] get_cpus is not supporting cgroup2
The current detection of https://github.com/conan-io/conan/blob/1.46.2/conans/client/tools/oss.py#L36-L39 seems no longer applicable in case of cgroup2 as the `/sys/fs/cgroup/cpu/cpu.cfs_quota_us` and `/sys/fs/cgroup/cpu/cpu.cfs_period_us` are replaced by `/sys/fs/cgroup/cpu.max` (in case of docker executed with `--cpus`).

There could be checked if cgroup2 is active for example by checking for existence of `/sys/fs/cgroup/cgroup.controllers` file and based on that the values for `quota` and `period` could be taken from `cpu.max` file.

### PR DESCRIPTION
**Pull request by MariaMozgunova**
Changelog: Fix: Get the cpu count for cgroup2 from `/sys/fs/cgroup/cpu.max`.
Docs: https://github.com/conan-io/docs/pull/2658

Fixes #11635

Gets the cpu count for cgroup2 from `/sys/fs/cgroup/cpu.max` file. From [docs](https://www.kernel.org/doc/html/latest/admin-guide/cgroup-v2.html), `/sys/fs/cgroup/cpu.max` file has the following format: `$MAX $PERIOD`.
In case `$MAX` is "max", return 1.
If there is no `$PERIOD`, use only `$MAX` as the return value.
When both values are integers, use them as before.

- [x] Refer to the issue that supports this Pull Request.
- [x] If the issue has missing info, explain the purpose/use case/pain/need that covers this Pull Request.
- [x] I've read the [Contributing guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [x] I've followed the PEP8 style guides for Python code.
- [x] I've opened another PR in the Conan docs repo to the ``develop`` branch, documenting this one. 

<sup>**Note:** By default this PR will skip the slower tests and will use a limited set of python versions. Check [here](https://github.com/conan-io/conan/blob/develop/.github/PR_INCREASE_TESTING.md) how to increase the testing level by writing some tags in the current PR body text.</sup>

### CODE REVIEW

**Comment by MariaMozgunova** [comment]
Should I document the changes to the Conan docs?

**Review by SSE4 — APPROVED** [review]

**Comment by memsharded** [comment]
> Should I document the changes to the Conan docs?

Better wait until it is merged, and then yes, a PR to the docs, if necessary, would be nice, just a couple of lines if it makes sense

**Comment by MariaMozgunova** [comment]
Do I need to do something for this PR to be merged?

**Review by memsharded — APPROVED** [review]

**Comment by memsharded** [comment]
> Do I need to do something for this PR to be merged?

No, it is good, it just takes some time to review, too many ongoing things. It doesn't have tests, but honestly all tests are going to be pure mocks, so probably it is good.

Maybe @maikelvdh want to comment something more, if not, this will be merged and released in 1.51 (still a few days, Conan 2.0-beta.2 is in the next release)

File: conan/tools/build/cpu.py
Comment (maikelvdh) [inline conan/tools/build/cpu.py]: In case of `max` it means basically no limits are established and therefore we should be able to rely on the `multiprocessing.cpu_count()`. In the current logic you would otherwise effectively in case of docker when cgroup V2 present and without any limiting of CPU cores applied just run with 1 jobs always.

File: conan/tools/build/cpu.py
Comment (maikelvdh) [inline conan/tools/build/cpu.py]: To keep it in line with the actual official documentation and the microsecond notation
```suggestion
                    cfs_quota_us, cfs_period_us = int(cpu_max[0]), 100000
```

**Review by maikelvdh — CHANGES_REQUESTED** [review]

File: conan/tools/build/cpu.py
Comment (memsharded) [reply conan/tools/build/cpu.py]: But is it not an issue that ``multiprocessing.cpu_count()`` is used? because it will not match the docker assigned cores, and it can also overflow the container capacity?

Comment (maikelvdh) [reply conan/tools/build/cpu.py]: That is exactly what you want in case of docker if you are not using the runtime options to limit the CPU (https://docs.docker.com/config/containers/resource_constraints/#cpu). In case you are using this options to limit you will get not `max` as value.

Comment (memsharded) [reply conan/tools/build/cpu.py]: Great, thanks very much for the clarification. @MariaMozgunova could you please apply this in your PR? Thanks!

**Comment by maikelvdh** [comment]
Thanks a lot @MariaMozgunova for addressing this problem 🚀 !

**Comment by memsharded** [comment]
Merged, thanks for your contribution! It will be part of 1.51

If you now want to do a PR to the docs, something very simple about the behavior for cgroups2, a couple of lines would be enough, that would be great.

**Comment by MariaMozgunova** [comment]
@memsharded Can you please hint to me what file precisely should be modified?

**Comment by uilianries** [comment]
@MariaMozgunova The documentation already describes cgroup usage, so with your hotfix, the follow part should updated:

https://github.com/conan-io/docs/blob/79feb32d0be4d531a363b8948b6d8a2289e059ab/reference/tools.rst#toolscpu_count

The website will look like this: https://docs.conan.io/en/latest/reference/tools.html#tools-cpu-count

Thank you for your contribution!

**Pull request merged**
